django/db/models/fields/__init__.py

```diff
@@ -2319,6 +2319,8 @@ def get_db_prep_value(self, value, connection, prepared=False):
 
     def to_python(self, value):
         if value is not None and not isinstance(value, uuid.UUID):
+            if isinstance(value, int):
+                return uuid.UUID(int=value)
```

Comment:
This needs to be done in the `ValueError` `try`/`except` below as this will fail for integer larger than 128 bits. A test should be added for this case as well.

```python
> uuid.UUID(int=(2 ** 128) - 1)
UUID('ffffffff-ffff-ffff-ffff-ffffffffffff')
> uuid.UUID(int=(2 ** 128))
ValueError: int is out of range (need a 128-bit value)
```

Reply:
Oh, yes! We should check that.

I've modified the patch to keep a check on these things. Can you have a look now, please?